### PR TITLE
RS-15247: avoid when too many bins are used in a histogram

### PR DESCRIPTION
**Pull request by chschan**

### CODE REVIEW

File: R/distribution.R [hunk 1]
```diff
@@ -380,40 +380,42 @@ Distribution <-   function(x,
...
+                bin.size <- (rng[2] - rng[1])/maximum.bins
+                default.bins <- FALSE
+            }
+            else if (length(x.sorted) < 10)
+            {
+                # Use smaller bins when there are only a few values. This avoids grouping
+                # values together if they are unevenly spaced inside larger range.
+                bin.size <- bin.min.size
+                default.bins <- FALSE
+            }
```
Comment (JustinCCYap) [inline R/distribution.R:413]: What happens in the else? Does plotly determine the bin.size?

File: R/distribution.R [hunk 2]
```diff
@@ -380,40 +380,42 @@ Distribution <-   function(x,
         density.color <- rep(density.color, n.variables)
     if (length(density.color) != n.variables)
         warning("The number of colors provided for shading the densities is not consistent with the number of variables.")
-
     # Histograms can be specified using 'maximum.bins' or 'size' (which overrides the former)
```
Comment (JustinCCYap) [inline R/distribution.R:383]: Is there still a `size` parameter?

File: R/distribution.R [hunk 3]
```diff
@@ -380,40 +380,42 @@ Distribution <-   function(x,
...
     # but otherwise use 'maximum.bins' because it tends to find more rounded breakpoints
     x.sorted <- sort(unique(unlist(x)))
     rng <- x.sorted[c(1, length(x.sorted))]
-    default.bins <- is.null(maximum.bins) || is.na(maximum.bins)
-    if (is.null(maximum.bins) || is.na(maximum.bins))
-        maximum.bins <- min(length(x.sorted), 50)
-    bin.min.size <- min(diff(x.sorted))
     if (density.type == "Histogram")
     {
+        bin.min.size <- min(diff(x.sorted))
```
Comment (JustinCCYap) [inline R/distribution.R:391]: I wonder if we should have done something different if the values are non-integer, to avoid tiny bin sizes.

**Review by JustinCCYap — APPROVED** [review]

File: R/distribution.R (hunk 3 above)
Comment (chschan) [reply R/distribution.R:393]: `bin.min.size` is not actually used unless the sample has only a few unique values. But we use it to identify when plotly's default might run into problems. Added a comment in [d1ef499](https://github.com/Displayr/flipStandardCharts/pull/64/commits/d1ef4995aa959850235be7795b887f7942a85de2) to explain this

File: R/distribution.R (hunk 1 above)
Comment (chschan) [reply R/distribution.R:417]: Yes. Added a comment to explain that

File: R/distribution.R (hunk 3 above)
Comment (JustinCCYap) [reply R/distribution.R:393]: Don't we use `bin.min.size` to compute `maximum.bins` on line 402? If `bin.min.size` is small then `maximum.bins` could be large.

**Pull request merged**